serializers/src/main/java/io/pravega/schemaregistry/serializers/MultiplexedSerializer.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.serializers;
+
+import io.pravega.client.stream.Serializer;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+class MultiplexedSerializer<T> implements Serializer<T> {
+    private final Map<Class<? extends T>, AbstractPravegaSerializer<T>> serializers;
+
+    MultiplexedSerializer(Map<Class<? extends T>, AbstractPravegaSerializer<T>> serializers) {
+        this.serializers = serializers;
+    }
+    
+    @Override
+    @SuppressWarnings("unchecked")
+    public ByteBuffer serialize(T obj) {
+        Class<? extends T> tClass = (Class<? extends T>) obj.getClass();
+        AbstractPravegaSerializer<T> serializer = serializers.get(tClass);
+        return serializer.serialize(obj);
```

Comment:
In the absence of headers how can this possibly be deserialized?

Reply:
Multiplexed deserializers require writer schema to be available. They cannot be used without encoding the header. 
they mandate the precondition:
```
        Preconditions.checkNotNull(writerSchema);
```

Reply:
I see, this is an internal class. I missed that.